src/batSet.ml

```diff
@@ -793,6 +818,18 @@ let remove x s = Concrete.remove Pervasives.compare x s
 
 let iter f s = Concrete.iter f s
 
+let at_exn i s = Concrete.at_exn i s
+
+(*$T
+  at_exn 0 (of_list [1;2]) == 1
+  at_exn 1 (of_list [1;2]) == 2
```

Comment:
I think you could have a test with a bit more elements in the set -- for example with 5 elements, picking an element that is neither the first nor the last to avoid edge cases.